src/glados/engine/tui_integration.py

```
@@ -0,0 +1,193 @@
+        # Notify TUI of the interruption with detailed information
+        self.notify_tui("tts_interruption", {
+            "original_text": generated_text,
+            "clipped_text": clipped_text,
+            "percentage": percentage_played,
+            "reason": "voice_activation",
+            "timestamp": time.strftime("%H:%M:%S")
+        })
+        
+        return clipped_text
+
+    def _process_llm_stream_chunk(self, chunk: str, reasoning_step: str = None) -> None:
```

Ruff (0.11.9)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)